src/gen/lib/AST_grammar.ml

```diff
@@ -4,10 +4,26 @@
 
 type ident = string
 
+type alias = {
+  id: ident;
+    (* unique name for the alias, suitable for generating type names and
+       function names. *)
+
+  name: ident;
+    (* original name of the alias, which may not be unique.
+       This is what's exposed in the tree-sitter parser output. *)
+}
+
 type rule_body =
   (* atomic (leaves) *)
-  | Symbol of ident
+  | Symbol of (ident * alias option)
+     (* reference to a named rule. The optional part is the name shown to us
+        in the parsing result due to an ALIAS. *)
+
   | String of string
+     (* constant string as exposed in the parsing output from tree-sitter.
+        It is either the result of a STRING or an anonymous ALIAS. *)
+
```

Comment:
maybe an example for the anonymous ALIAS, like saying "(as in alias($._binary_minus, '-') in Ruby's grammar.js)"